### PR TITLE
Lib refactoring

### PR DESCRIPTION
**Pull request by gregdel**
* Refactoring of the general names for a better interface
* Add functional tests for the movies in the library
* Move the NFO into a separate package
* Add ShowSeason struct
* Add seasons and episode list in HTTP calls
* Update HTTP route declaration format
* Update HTTP route names
* Update the show routes to be more flexible in the future
* Update the pam module (detailer)

### CODE REVIEW

**Comment by gregdel** [comment]
The builds will fail until we merge https://github.com/odwrtw/papi/pull/2

File: lib/library/library.go [hunk 1]
```diff
@@ -252,13 +222,17 @@ func (l *Library) AddShowEpisode(ep *polochon.ShowEpisode, log *logrus.Entry) er
...
 
 	if !exists(showNFOPath) {
-
-		show := polochon.NewShowFromEpisode(ep)
-		if err := show.GetDetails(log); err != nil {
-			errors.LogErrors(log, err)
-			if errors.IsFatal(err) {
-				return err
+		show := ep.Show
+		pretty.Println(show)
```
Comment (PouuleT) [inline lib/library/library.go]: woups

File: lib/library/functional_movies_test.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	}
+	movieCount = len(ids)
+	if movieCount != 0 {
+		t.Fatalf("the library should contains %d movie instead of 0", movieCount)
+	}
+
+	// Ensure the movie folder has been deleted
+	if exists(lib.getMovieDir(m)) {
+		t.Fatal("the movie directory should have been deleted", err)
+	}
```
Comment (PouuleT) [inline lib/library/functional_movies_test.go]: We should also test that after a RebuildIndex, everything's still OK

File: lib/library/functional_shows_test.go [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+		t.Fatalf("failed to remove the episode: %q", err)
+	}
+
+	// Ensure the index if valid
+	gotIDs := lib.ShowIDs()
+	expectedIDs := map[string]index.IndexedShow{}
+
+	if !reflect.DeepEqual(expectedIDs, gotIDs) {
+		t.Errorf("invalid show ids, expected %+v got %+v", expectedIDs, gotIDs)
+	}
```
Comment (PouuleT) [inline lib/library/functional_shows_test.go]: Same here, test to rebuild the index and double check =)

**Pull request merged**